develop/tutorials/articles/110-portlets/15-javascript/04-npm/08-npm-sdk.markdown

```diff
@@ -1,171 +1,208 @@
+*feature level* refers to a complete set of component versions that produces 
+stable features(described by LPSs and GitHub issues). Feature levels are defined 
+by the minimum component versions needed for the supported features to work. 
+Note that you can use higher versions in any of the components, as long as they 
+are compatible with the one defined by the feature level.
 
-* gradle: 12.0
-* gradle plugin for node: 17.0.0
-* npm: 8.0
-* babel: 21.0
-* liferay-npm-build-tools: 27.0.0
-* frontend-js-loader-modules-extender: 16.0.0
-* liferay-amd-loader: 8.0.0
-
-We are no longer in feature level 8, as we have introduced the possibility of a
-breaking change when updating frontend-js-loader-modules-extender from 15.0.0
-to 16.0.0. 
-
-So, where are we if we said that feature level 8 was the maximum possible? In 
-this case we would be in an indeterminate feature level. Probably all features
-of level 8 would work, but we cannot assure it with 100% confidence because no
-feature level 9 has yet been released, so nobody has tested that the given 
-versions combination works correctly. This is the typical case of using a 
-development/unstable version of the SDK which has not yeet been released 
-officially. Thus, we could call this indeterminate state, feature level pre-9 or
-something alike.
-
-## Usage
-
-The tool is a CLI driven by commands given as arguments (much in the same way as
-tools like git or blade). 
-
-To see the list of all supported commands run `lnka help`. Then, to obtain help
-about a specific command run `lnka <command> help`.
-
-Detailed descriptions of each supported command follow in the next sections.
-
-### features
-
-The `lnka features` command can be run in a project folder that uses the npm SDK 
-to determine its feature level. 
-
-When this command is run it examines the versions of all related components and
-uses them to report the supported feature level.
-
-Keep in mind that the tool may not be able, for different reasons, to retrieve 
-the version number of a component. In that case the reported feature level may 
-be higher than the real one.
-
-For example: the tool needs to contact Liferay Portal through HTTP to get the 
-AMD loader version and through the GoGo console to get the 
-frontend-js-loader-modules-extender version. Thus, if you don't have a Portal 
-available, the tools cannot determine the version number for those components
-and does not take them into account when calculating the feature level.
-
-### man
-
-The `lnka man <item>` command shows information about the requested `<item>`. 
-Use `lnka man` (with no item) to get a list of supported items.
++$$$
 
+**Note:** Certain 
+[blade samples](https://github.com/liferay/liferay-blade-samples/tree/master/gradle/apps/npm)
+and 
+[templates](https://github.com/liferay/liferay-portal/tree/master/modules/sdk/project-templates) 
+may rely on a specific feature level.
+
```

Comment:
@izaera 
How can you determine what feature level is required for the template or sample to work?
Running this tool will list the available feature levels and tell which one you have, but it doesn't tell you what feature level is required to run your project.

Reply:
There's no way to know that from a tool, because only the author knows what the project does and what features it is using. 

That said, the only one who knows what a blade sample is using is the author of the blade sample. Maybe we could add some kind of readme to the sample, or any similar thing, but in the end it is a problem of blade samples theirselves because you have the same uncertainty with other samples: you cannot know, for example, what FP or GA is needed for a certain sample to run.

So, in the end, this problem should be fixed by the blade samples repo as a whole IMO.

Reply:
Ok. That makes sense. Thanks!